controllers/glance_controller.go

```diff
@@ -266,7 +265,7 @@ func (r *GlanceReconciler) reconcileDelete(ctx context.Context, instance *glance
 	r.Log.Info(fmt.Sprintf("Reconciling Service '%s' delete", instance.Name))
 
 	// remove db finalizer first
-	db, err := mariadbv1.GetDatabaseByName(ctx, helper, instance.Name)
+	db, err := mariadbv1.GetDatabaseByNameAndAccount(ctx, helper, instance.Name, instance.Spec.DatabaseAccount, instance.Namespace)
```

Comment:
If the mechanism provided by this patch will be the default, should this function simple be `GetDatabase(...` ?

Reply:
not really sure, I find the Database facade kind of awkward, because it seems like it wants to act like an object you'd "get", but it's more like a package of things that have to be loaded together.     I guess that's why an awkward function name seems fitting at the moment :)